tests/cpptests/test_cpp_query.cpp

```diff
@@ -512,6 +512,37 @@ TEST_F(QueryTest, testParser_v2) {
   assertInvalidQuery("(hello world => [KNN 10 @vec_field $BLOB]) @title:other", ctx);
   assertInvalidQuery("hello world => [KNN 10 @vec_field $BLOB] OR other => [KNN 10 @vec_field $BLOB]", ctx);
 
+  // Test range queries
+  assertValidQuery("@v:[RANGE 0.01 $BLOB]", ctx);
+  assertValidQuery("@v:[RANGE 2 $BLOB]", ctx);
+  assertValidQuery("@v:[RANGE $radius $BLOB]", ctx);
+  assertValidQuery("@v:[RANGE 2e-2 $BLOB]", ctx);
+  assertValidQuery("@v:[RANGE 2E-2 $BLOB]", ctx);
+  assertValidQuery("@v:[RANGE 0.01 $BLOB AS V_SCORE]", ctx);
+  assertValidQuery("@v:[RANGE 0.01 $BLOB AS __v_score]", ctx);
+  assertValidQuery("@v:[RANGE 0.01 $BLOB AS $V_SCORE]", ctx);
+  assertValidQuery("@v:[RANGE 0.01 $BLOB epsilon 0.1]", ctx);
+  assertValidQuery("@v:[RANGE 0.01 $BLOB epsilon $ep]", ctx);
+  assertValidQuery("@v:[RANGE 0.01 $BLOB epsilon 0.1 AS V_SCORE]", ctx);
+  assertValidQuery("@v:[RANGE 0.01 $BLOB epsilon $ep AS V_SCORE]", ctx);
+  assertValidQuery("@v:[RANGE 0.01 $BLOB epsilon 0.1 AS $V_SCORE]", ctx);
+  assertValidQuery("@v:[RANGE $r $BLOB epsilon $ep AS $V_SCORE]", ctx);
+
+  // Complex queries with range
+  assertValidQuery("@v:[RANGE 0.01 $BLOB] @text:foo OR bar", ctx);
+  assertValidQuery("(@v:[RANGE 0.01 $BLOB] @text:foo) => { $weight: 2.0 }", ctx);
+  assertValidQuery("@v:[RANGE 0.01 $BLOB] @text:foo OR bar @v:[RANGE 0.04 $BLOB2]", ctx);
+  assertValidQuery("(@v:[RANGE 0.01 $BLOB] @text:foo) => [KNN 5 @v $BLOB2]", ctx);
+  assertValidQuery("@v:[RANGE 0.01 $BLOB AS first_score] => [KNN 5 @v2 $BLOB2 AS second_score]", ctx);
+
+  // Invalid queries
+  assertInvalidQuery("@v:[BAD 0.01 $BLOB]", ctx);
+  assertInvalidQuery("@v:[RANGE 0.01]", ctx);
+  assertInvalidQuery("@v:[RANGE $BLOB]", ctx);
+  assertInvalidQuery("@v:[RANGE bad $BLOB]", ctx);
+  assertInvalidQuery("@v:[RANGE 0.01 notParam]", ctx);
+  assertInvalidQuery("@v:[RANGE 0.01 paramNameWithoutVal $BLOB]", ctx);
```

Comment:
Are there tests for RANGE queries with param substitution? (could be in python)

Reply:
I'll add some